zcash_client_backend/src/keys.rs

```diff
@@ -593,7 +627,7 @@ impl UnifiedFullViewingKey {
 
         // Find a working diversifier and construct the associated address.
         loop {
-            let res = self.address(j);
+            let res = self.address(j, receiver_types);
```

Comment:
`self.address()` currently returns `None` only if the diversifier index is invalid for the receiver types (previously implied by the `UnifiedFullViewingKey` components, and now explicitly stated), which means that incrementing `j` below is the correct behaviour here.

If we make my change above to validate `receiver_types` then we will also have to either pre-validate `receiver_types` in `find_address` before this call, or change the return type of `self.address()` to distinguish these two cases. Otherwise we'll get stuck in a seemingly-infinite loop here (because an unsupported receiver type will be invalid for all `j`).

Reply:
See also https://github.com/zcash/librustzcash/pull/829#discussion_r1442391323